core/lib/spree/service_module.rb

```diff
@@ -0,0 +1,64 @@
+module Spree
+  module ServiceModule
+    class MethodNotImplemented < StandardError; end
+    class WrongDataPassed < StandardError; end
+    class NonCallablePassedToRun < StandardError; end
+    class CallMethodNotImplemented < StandardError; end
+
+    Result = Struct.new(:success, :value) do
+      def success?
+        success
+      end
```

Comment:
Could we move it to a separate gem?

Reply:
@daniilbober I'm huge fan of whole dry-rb but for me, dry-transaction is too feature-heavy and overkill for such simple task. I want to keep it as simple as possible. dry-transaction, due to having step adapters, step notifications, encouraging to use containers introduces a lot of inconsistency and I'd like to keep this as it is now. 

I'll be testing how it works with spree and maybe extend it if needed.

Reply:
Yea, that makes a sense. And it also can be obscure for programmers who didn't with all this stuff. Anyway the direction of refactoring that you chose is really interesting to keep track of, and generally is much nicer than previous versions for me